src/halmos/sevm.py

```diff
@@ -2097,17 +2094,11 @@ def finalize(ex: Exec):
                 if ex.context.depth > MAX_CALL_DEPTH:
                     raise MessageDepthLimitError(ex.context)
 
-                # print(f"{hexify(ex.this)}@{ex.pc}", end=" ")
                 insn = ex.current_instruction()
-                # print(f"insn={insn}")
-
                 opcode = insn.opcode
-                ex.cnts["opcode"][opcode] += 1
 
-                if (
-                    "max_depth" in self.options
-                    and sum(ex.cnts["opcode"].values()) > self.options["max_depth"]
-                ):
+                max_depth = self.options.get("max_depth", 0)
+                if max_depth and step_id > max_depth:
```

Comment:
unlike cnts, step_id is intra-procedural, not accumulating the steps of external callees.

to make this optimization correct, we need to have step_id to be accumulated across external calls.

Reply:
not sure what you mean, I see step_id incrementing monotonically along a single path. This may slightly change the meaning of `max_depth`, but the new behavior sounds reasonable to me

<img width="820" alt="image" src="https://github.com/a16z/halmos/assets/85039585/cfb13d9f-97e5-460b-ae90-cfeccc201dbf">

Reply:
oops, sorry my bad. a separate sevm is no longer spawn for external calls, since the new dfs path exploration. so yes, step_id increases across external calls. great catch, thanks!